device/cuda/CMakeLists.txt

```diff
@@ -1,53 +1,53 @@
-  "src/utils/make_prefix_sum_buff.cu"
-  "include/traccc/cuda/utils/stream.hpp"
-  "src/utils/stream.cpp"
-  "src/utils/opaque_stream.hpp"
-  "src/utils/opaque_stream.cpp"
-  "src/utils/utils.hpp"
-  "src/utils/utils.cpp"
-  # Seed finding code.
-  "include/traccc/cuda/seeding/track_params_estimation.hpp"
-  "include/traccc/cuda/seeding/seed_finding.hpp"
-  "include/traccc/cuda/seeding/seeding_algorithm.hpp"
-  "include/traccc/cuda/seeding/spacepoint_binning.hpp"
-  # CCL code.
-  "include/traccc/cuda/cca/component_connection.hpp"
-  "src/seeding/track_params_estimation.cu"
-  "src/seeding/seed_finding.cu"
-  "src/seeding/spacepoint_binning.cu"
-  "src/seeding/seeding_algorithm.cpp"
-  "src/cca/component_connection.cu"
-  # Clusterization
-  "include/traccc/cuda/clusterization/clusterization_algorithm.hpp"
-  "src/clusterization/clusterization_algorithm.cu"
-  # Fitting
-  "include/traccc/cuda/fitting/fitting_algorithm.hpp"
-  "src/fitting/fitting_algorithm.cu")
-  
-target_compile_options( traccc_cuda
-  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr> )
-target_link_libraries( traccc_cuda
-  PUBLIC traccc::core detray::core detray::utils vecmem::core covfie::core
-  PRIVATE CUDA::cudart traccc::device_common vecmem::cuda )
+#Set up the build of the traccc::cuda library.
+traccc_add_library(traccc_cuda cuda TYPE SHARED
+#Utility definitions.
+        "include/traccc/cuda/utils/definitions.hpp"
+        "include/traccc/cuda/utils/make_prefix_sum_buff.hpp"
+        "src/utils/make_prefix_sum_buff.cu"
+        "include/traccc/cuda/utils/stream.hpp"
+        "src/utils/stream.cpp"
+        "src/utils/opaque_stream.hpp"
+        "src/utils/opaque_stream.cpp"
+        "src/utils/utils.hpp"
+        "src/utils/utils.cpp"
+#Seed finding code.
+        "include/traccc/cuda/seeding/track_params_estimation.hpp"
+        "include/traccc/cuda/seeding/seed_finding.hpp"
+        "include/traccc/cuda/seeding/seeding_algorithm.hpp"
+        "include/traccc/cuda/seeding/spacepoint_binning.hpp"
+#CCL code.
+        "include/traccc/cuda/cca/component_connection.hpp"
+        "src/seeding/track_params_estimation.cu"
+        "src/seeding/seed_finding.cu"
+        "src/seeding/spacepoint_binning.cu"
+        "src/seeding/seeding_algorithm.cpp"
+        "src/cca/component_connection.cu"
+#Clusterization
+        "include/traccc/cuda/clusterization/clusterization_algorithm.hpp"
+        "src/clusterization/clusterization_algorithm.cu"
+#Fitting
+        "include/traccc/cuda/fitting/fitting_algorithm.hpp"
+        "src/fitting/fitting_algorithm.cu")
 
+target_compile_options(traccc_cuda PRIVATE
+        $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr
+        $<$<BOOL:${TRACCC_SETUP_PROFILER}>:-Xcompiler -finstrument-functions -fPIC>>)
+target_link_libraries(traccc_cuda PUBLIC
+        traccc::core detray::core detray::utils vecmem::core covfie::core
+        PRIVATE CUDA::cudart traccc::device_common vecmem::cuda nvToolsExt)
```

Comment:
Please ensure that `nvToolsExt` is only linked against if `TRACCC_SETUP_PROFILER` is enabled, using the same logic as you did above (using the generator expression).

Reply:
Just tried running the code without `nvToolsExt` . Looks like it was not necessary

Reply:
ok I added the line:
`INTERFACE $<$<BOOL:${TRACCC_SETUP_PROFILER}>: CUDA::nvToolsExt`

seems to be working

Reply:
Great, thanks!